code/android-signal-library/src/main/java/com/adobe/marketing/mobile/signal/SignalHitProcessor.kt

```diff
@@ -92,7 +87,7 @@ internal class SignalHitProcessor : HitProcessing {
             countDownLatch.countDown()
         }
         countDownLatch.await((request.connectTimeout + 1).toLong(), TimeUnit.SECONDS)
-        return result
+        processingResult.complete(result)
```

Comment:
Can you avoid the latch and call `processingResult.complete(result)` inside the http handler?